Setting --output-root ${CASEDIR} and --case ${CASEDIR}/${CASENAME} results in MCT build error as of 5.8.0 on Cheyenne

Not sure if this is desired or not, but figured I'd note it.

Previously, when doing dev work w/ CAM/CESM I set:
--case ${CASEDIR}/${CASENAME} --output-root ${CASEDIR}

... during the create_newcase phase so I could have "self-contained" directories on my scratch space for playing around in (rather than splitting them between home/scratch like I normally do for production runs)

This seems to be acceptable up until cesm2_2_beta01 (CIME 5.7.x)...

As of cesm2_2_beta02 (>=CIME 5.8) this setup results in an error when building mct libraries...

```
/glade/u/home/zarzycki/scratch/TC_PBL/ne30test/Tools/mkSrcfiles
cp -f Filepath Deppath
/glade/u/home/zarzycki/scratch/TC_PBL/ne30test/Tools/mkDepends  Deppath Srcfiles > Depends
gmake: Leaving directory '/glade/scratch/zarzycki/TC_PBL/ne30test/bld/intel/mpt/nodebug/nothreads/mct/mct'
cat: Filepath: No such file or directory
cat: Srcfiles: No such file or directory
gmake: *** No rule to make target '/glade/u/home/zarzycki/scratch/TC_PBL/ne30test/bld/intel/mpt/nodebug/nothreads/mct/mct/Makefile.conf'.  Stop.
ERROR: cat: Filepath: No such file or directory
cat: Srcfiles: No such file or directory
gmake: *** No rule to make target '/glade/u/home/zarzycki/scratch/TC_PBL/ne30test/bld/intel/mpt/nodebug/nothreads/mct/mct/Makefile.conf'.  Stop.
```

The way to get around this (for me) is to have the --output_root be external to the case dir, this seems to allow MCT to build fine and the rest of the components to proceed as expected. Given this, I assume there is some relative path issue with ../ or similar?

This can be easily reproduced on Cheyenne w/ default sys modules by checking out cesm2_2_beta02, configuring a generic F compset, and running ./case.setup, ./case.build.

--- a/scripts/lib/CIME/case/case.py
+++ b/scripts/lib/CIME/case/case.py
@@ -1021,6 +1021,7 @@ class Case(object):
         # we don't want to resolve variables until we need them
         if output_root is None:
             output_root = self.get_value("CIME_OUTPUT_ROOT")
+        output_root = os.path.abspath(output_root)
         self.set_value("CIME_OUTPUT_ROOT", output_root)
         if non_local:
             self.set_value("EXEROOT", os.path.join(output_root, self.get_value("CASE"), "bld"))

